app/models/cti/caller_id.rb

```diff
@@ -197,40 +199,21 @@ def self.config
 =end
 
     def self.parse_text(text)
-      caller_ids = []
-
-      # 022 1234567
-      # 021 123 2345
-      # 0271233211
-      # 021-233-9123
-      # 09 123 32112
-      # 021 2331231 or 021 321123123
-      # 622 32281
-      # 5754321
-      # 092213212
-      # (09)1234321
-      # +41 30 53 00 00 000
-      # +42 160 0000000
-      # +43 (0) 30 60 00 00 00-0
-      # 0043 (0) 30 60 00 00 00-0
-
-      default_country_id = '49'
-      text.gsub!(/([\d|\s|\-|\(|\)]{6,26})/) {
-        number = $1.strip
-        number.sub!(/^00/, '')
-        number.sub!(/\(0\)/, '')
-        number.gsub!(/(\s|\-|\(|\))/, '')
-        if !Phony.plausible?(number)
-          if number =~ /^0/
-            number.gsub!(/^0/, default_country_id)
-          else
-            number = "#{default_country_id}#{number}"
-          end
-          next if !Phony.plausible?(number)
+      # see specs for example
+      text.scan(/([\d|\s|\-|\(|\)]{6,26})/).map do |match|
```

Comment:
I think the yielded block should be moved into a separate function since here are (at least) two functionalities in one method:

1. looping over a text extracting numbers
2. working up numbers to be in a specific format

You can see this even clearer in the tests where those two get mixed together.

Would be great if you could split this up. This would make the tests and API more clearer and splits the responsibilities as they should be. In the long term both functions could be used independently for various use cases.